flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,285 @@
+                        || (exp.getField(0).equals(value.getField(1)));
+            }
+        }
+    }
+
+    /** test knn model load and transform. */
+    @Test
+    public void testEstimatorLoadAndSave() throws Exception {
+        String path = Files.createTempDirectory("").toString();
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("vec")
+                        .setK(4)
+                        .setPredictionCol("pred");
+        knn.save(path);
+
+        Knn loadKnn = Knn.load(path);
+        KnnModel knnModel = loadKnn.fit(trainData);
+        Table result = knnModel.transform(testData)[0];
+
+        DataStream<Row> output = tEnv.toDataStream(result);
+
+        List<Row> rows = IteratorUtils.toList(output.executeAndCollect());
+        for (Row value : rows) {
+            for (Row exp : expectedData) {
+                assert !exp.getField(1).equals(value.getField(0))
+                        || (exp.getField(0).equals(value.getField(1)));
+            }
+        }
+    }
+
+    /** Test knn model load and transform. */
+    @Test
+    public void testModelLoadAndSave() throws Exception {
+        String path = Files.createTempDirectory("").toString();
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("vec")
+                        .setK(4)
+                        .setPredictionCol("pred");
+        KnnModel knnModel = knn.fit(trainData);
+        knnModel.save(path);
+        env.execute();
+
+        KnnModel newModel = KnnModel.load(env, path);
+        Table result = newModel.transform(testData)[0];
+
+        DataStream<Row> output = tEnv.toDataStream(result);
+
+        List<Row> rows = IteratorUtils.toList(output.executeAndCollect());
+        for (Row value : rows) {
+            for (Row exp : expectedData) {
+                assert !exp.getField(1).equals(value.getField(0))
+                        || (exp.getField(0).equals(value.getField(1)));
+            }
+        }
+    }
+
+    /** Test Param. */
+    @Test
+    public void testParam() {
+        Knn knnOrigin = new Knn();
+        assertEquals("label", knnOrigin.getLabelCol());
+        assertEquals(10L, knnOrigin.getK().longValue());
+        assertEquals("prediction", knnOrigin.getPredictionCol());
+
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("vec")
+                        .setK(4)
+                        .setPredictionCol("pred");
+
+        assertEquals("vec", knn.getFeaturesCol());
+        assertEquals("label", knn.getLabelCol());
+        assertEquals(4L, knn.getK().longValue());
+        assertEquals("pred", knn.getPredictionCol());
+    }
+
+    /** Test model data. */
+    @Test
+    public void testModelData() throws Exception {
```

Comment:
It will be nice to make the tests structure consistent with other algorithms (see e.g. `KMeansTest` or tests in other pending PRs). If there is good reason to add/remove/modify the test structure, we can apply that to every algorithm.

Following this logic, maybe we can rename this test as `testGetModelData` and add a `testSetModelData`.

Could you go over all tests in this class and see if there is anything that could be improved?

Reply:
I will refine the ut as others later.

Reply:
done